documentation/src/crates/primitives/kzg.md

```diff
@@ -0,0 +1,30 @@
+# KZG 
+
+With the introduction of EIP4844, this use of blobs for a more efficent short term storage is employed, the validity of this blob stored in the consensus layer is verified 
+using the `Point Evaluation` pre-compile, a fancy way of verifing that and evaluation at a given point of a commited polynomial is vaild, in a much more bigger scale, implies that `Data is Available`.
+
+This module houses;
+
+1. `KzgSettings`: Stores the setup and parameters needed for computing KZG proofs.
+
+    ```sh
+        pub struct KZGSettings {
+            #[doc = " The length of `roots_of_unity`, a power of 2."]
+            max_width: u64,
+            #[doc = " Powers of the primitive root of unity determined by\n `SCALE2_ROOT_OF_UNITY` in bit-reversal permutation order,\n length `max_width`."]
+            roots_of_unity: *mut fr_t,
+            #[doc = " G1 group elements from the trusted setup,\n in Lagrange form bit-reversal permutation."]
+            g1_values: *mut g1_t,
+            #[doc = " G2 group elements from the trusted setup."]
+            g2_values: *mut g2_t,
+        }
+    ```
```

Comment:
No need to share Struct here, but it needs to be mentioned that we have default KzgSettings that is obtained from kzg ceremony and we allow setting custom KzgSetting (if needed) inside `env.cfg`

Would be good to link kzg ceremony and even the EIP itself. For example https://ceremony.ethereum.org/ and https://eips.ethereum.org/EIPS/eip-4844